### PR TITLE
Fix/Update plugins

### PR DESCRIPTION
**Pull request by nrobinaubertin**
### Overview

- sorted some lists of instructions/variables
- fixed the commit for all the plugins to target the latest compatible with our discourse version
- switch the mermaid plugin to the official one
- add a missing dependency for the `setup` step on the `discourse-calendar` step
- remove some unnecessary patches

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The changes are compliant with [ISD054 - Manging Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`
- [x] The documentation for charmhub is updated.
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

File: discourse_rock/rockcraft.yaml [hunk 1]
```diff
@@ -39,10 +39,10 @@ parts:
     build-environment:
       - ARCH: "x64"
       - NODE_VERSION: "18.18.2"
-      - YARN_VERSION: "1.22.19"
-      - RUBY_VERSION: "3.2.2"
-      - RUBY_INSTALL_VERSION: "0.9.2"
       - RAILS_ENV: "production"
+      - RUBY_INSTALL_VERSION: "0.9.2"
```
Comment (merkata) [inline discourse_rock/rockcraft.yaml:43]: You did the vim sort, didn't you.

Comment (nrobinaubertin) [reply discourse_rock/rockcraft.yaml:43]: Yes, is that bad ? (`:sort lu`)

Comment (merkata) [reply discourse_rock/rockcraft.yaml:43]: On the contrary, it's great!

**Review by merkata — DISMISSED** [review]

File: discourse_rock/rockcraft.yaml [hunk 2]
```diff
@@ -66,18 +66,14 @@ parts:
     source-type: git
     override-build: |
       craftctl default
-      mkdir -p tmp/backups/default
+      mkdir -p log/production.log
```
Comment (arturo-seijas) [inline discourse_rock/rockcraft.yaml:69]: Is this supposed to be a `touch`?

File: discourse_rock/rockcraft.yaml [hunk 3]
```diff
@@ -221,28 +220,28 @@ parts:
...
-      bin/bundle install --gemfile="plugins/discourse-saml/Gemfile"
+      bin/bundle install --gemfile="plugins/discourse-calendar/Gemfile"
       bin/bundle install --gemfile="plugins/discourse-data-explorer/Gemfile"
+      bin/bundle install --gemfile="plugins/discourse-gamification/Gemfile"
       bin/bundle install --gemfile="plugins/discourse-prometheus/Gemfile"
+      bin/bundle install --gemfile="plugins/discourse-saml/Gemfile"
       bin/bundle install --gemfile="plugins/discourse-solved/Gemfile"
       bin/bundle install --gemfile="plugins/discourse-templates/Gemfile"
-      bin/bundle install --gemfile="plugins/discourse-calendar/Gemfile"
-      bin/bundle install --gemfile="plugins/discourse-gamification/Gemfile"
```
Comment (arturo-seijas) [inline discourse_rock/rockcraft.yaml:230]: Is this removal intentional?

Comment (nrobinaubertin) [reply discourse_rock/rockcraft.yaml:230]: It's just a reorder of the lines

File: discourse_rock/rockcraft.yaml (hunk 2 above)
Comment (nrobinaubertin) [reply discourse_rock/rockcraft.yaml:69]: There's a `touch` for it line 75

Comment (arturo-seijas) [reply discourse_rock/rockcraft.yaml:69]: Shouldn't this be `mkdir -p log`?

File: discourse_rock/rockcraft.yaml (hunk 3 above)
Comment (arturo-seijas) [reply discourse_rock/rockcraft.yaml:230]: Indeed

**Review by arturo-seijas — APPROVED** [review]

**Review by amandahla — APPROVED** [review]
LGTM
